dialog/file.go

```diff
@@ -245,27 +294,82 @@ func showFileDialog(save bool, callback interface{}, parent fyne.Window) {
+// SetFilter sets a filter for limiting files that can be chosen in the file dialog
+func (f *File) SetFilter(filter FileFilter) {
+	f.filter = filter
+	if f.dialog != nil {
+		f.dialog.refreshDir(f.dialog.dir)
+	}
+}
+
+// Show shows the file dialog
+func (f *File) Show() {
+	if !f.save {
+		if fileOpenOSOverride(f.callback.(func(fyne.FileReadCloser, error)), f.parent) {
+			return
+		}
+		f.dialog = showFile(f)
+		return
+	}
+	if fileSaveOSOverride(f.callback.(func(fyne.FileWriteCloser, error)), f.parent) {
+		return
+	}
+	f.dialog = showFile(f)
+}
+
+// NewExtensionFileFilter takes a string slice of extensions with a leading . and creates a filter for the file dialog.
+// Example: .jpg, .mp3, .txt, .sh
+func NewExtensionFileFilter(extensions []string) FileFilter {
+	return &extensionFileFilter{extensions: extensions}
+}
+
+// NewMimeTypeFileFilter takes a string slice of mimetypes, including globs, and creates a filter for the file dialog.
+// Example: image/*, audio/mp3, text/plain, application/*
+func NewMimeTypeFileFilter(mimeTypes []string) FileFilter {
+	return &mimeTypeFileFilter{mimeTypes: mimeTypes}
+}
+
+// NewFileOpen creates a file dialog allowing the user to choose a file to open.
+// The dialog will appear over the window specified when Show() is called.
+func NewFileOpen(callback func(fyne.FileReadCloser, error), parent fyne.Window) *File {
+	dialog := &File{callback: callback, parent: parent}
+	return dialog
+}
+
+// NewFileSave creates a file dialog allowing the user to choose a file to save to (new or overwrite).
+// If the user chooses an existing file they will be asked if they are sure.
+// The dialog will appear over the window specified when Show() is called.
+func NewFileSave(callback func(fyne.FileWriteCloser, error), parent fyne.Window) *File {
+	dialog := &File{callback: callback, parent: parent, save: true}
+	return dialog
+}
+
+// ShowFileOpen creates and shows a file dialog allowing the user to choose a file to open.
 // The dialog will appear over the window specified.
-func ShowFileOpen(callback func(fyne.FileReadCloser, error), parent fyne.Window) {
+func ShowFileOpen(callback func(fyne.FileReadCloser, error), parent fyne.Window) *File {
+	dialog := NewFileOpen(callback, parent)
 	if fileOpenOSOverride(callback, parent) {
-		return
+		return nil
 	}
-	showFileDialog(false, callback, parent)
+	dialog.Show()
+	return dialog
 }
 
-// ShowFileSave shows a file dialog allowing the user to choose a file to save to (new or overwrite).
+// ShowFileSave creates and shows a file dialog allowing the user to choose a file to save to (new or overwrite).
 // If the user chooses an existing file they will be asked if they are sure.
 // The dialog will appear over the window specified.
-func ShowFileSave(callback func(fyne.FileWriteCloser, error), parent fyne.Window) {
+func ShowFileSave(callback func(fyne.FileWriteCloser, error), parent fyne.Window) *File {
```

Comment:
The `Show*` methods I already know are just fire and forget and do not return the created entity.
It this about to change?
If so, why?
If not, why do the `ShowFile*` functions return the `File`?

Reply:
Good catch, I missed this - the Show functions should indeed not return the type, it's fire and forget as pointed out.